catalog/migrate/migrate.go

```diff
@@ -0,0 +1,333 @@
+package migrate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/treeverse/lakefs/catalog"
+	"github.com/treeverse/lakefs/catalog/mvcc"
+	"github.com/treeverse/lakefs/catalog/rocks"
+	"github.com/treeverse/lakefs/config"
+	"github.com/treeverse/lakefs/db"
+	"github.com/treeverse/lakefs/graveler"
+)
+
+type Migrate struct {
+	db            db.Database
+	mvccCataloger catalog.Cataloger
+	entryCatalog  *rocks.EntryCatalog
+	repositoryRe  *regexp.Regexp
+	// per repository state
+	lastCommit graveler.CommitID
+	branches   map[graveler.BranchID]graveler.CommitID
+	tags       map[graveler.TagID]graveler.CommitID
+}
+
+type commitRecord struct {
+	BranchID              int64
+	BranchName            string
+	CommitID              int64
+	PreviousCommitID      int64
+	Committer             string
+	Message               string
+	CreationDate          time.Time
+	Metadata              catalog.Metadata
+	MergeType             string
+	MergeSourceBranch     *int64
+	MergeSourceBranchName string
+	MergeSourceCommit     *int64
+}
+
+const (
+	migrateFetchSize = 1000
+
+	initialCommitter     = "migrate"
+	initialCommitMessage = "repository created"
+)
+
+func (m *Migrate) Close() error {
+	return m.mvccCataloger.Close()
+}
+
+func NewMigrate(db db.Database, cfg *config.Config) (*Migrate, error) {
+	entryCatalog, err := rocks.NewEntryCatalog(cfg, db)
+	if err != nil {
+		return nil, err
+	}
+
+	mvccParams := cfg.GetMvccCatalogerCatalogParams()
+	mvccCataloger := mvcc.NewCataloger(db, mvcc.WithParams(mvccParams))
+	return &Migrate{
+		db:            db,
+		entryCatalog:  entryCatalog,
+		mvccCataloger: mvccCataloger,
+	}, nil
+}
+
+func (m *Migrate) FilterRepository(expr string) error {
+	if expr == "" {
+		m.repositoryRe = nil
+		return nil
+	}
+	repositoryRe, err := regexp.Compile(expr)
+	if err != nil {
+		return err
+	}
+	m.repositoryRe = repositoryRe
+	return nil
+}
+
+func (m *Migrate) Run() error {
+	ctx := context.Background()
+	// logger := logging.FromContext(ctx)
+	fmt.Println("==> Listing repositories to import")
```

Comment:
I know this is a CLI, but can we log?  Alternatively, you could write to `fmt.Print*` under `cmd`, I guess (pass a "log it" callback to this function).

Reply:
will use logger and aggregate the errors for the cli